packages/runtime/src/next/utils/pagination.ts

```diff
@@ -0,0 +1,54 @@
+import { AsyncIterableExt } from './async-iterable'
+
+type PaginationResult<T> = { data: T[]; hasMore: boolean }
+type IterablePaginationResult<T> = Promise<PaginationResult<T>> & AsyncIterableExt<T>
+
+type PaginationFunction<Options, R extends { id: any }> = Options extends {
+  after?: string
+}
+  ? (options: Options) => Promise<PaginationResult<R>>
+  : never
+
+type IterableMethod<Options, R> = Partial<Options> extends Options
+  ? (options?: Options) => IterablePaginationResult<R>
+  : (options: Options) => IterablePaginationResult<R>
+
+export function toIterablePaginationResult<Options, R extends { id: any }>(
+  fn: PaginationFunction<Options, R>,
+): IterableMethod<Options, R> {
+  return (options?: Options): IterablePaginationResult<R> => {
+    return (() => {
```

Comment:
Is the IIFE necessary here?

Reply:
Good question, I don't think it is anymore since we already have an outer function scope where we can store the state.